src/main/java/online/talkandtravel/facade/impl/AuthenticationFacadeImpl.java

```diff
@@ -0,0 +1,163 @@
+   * @param password The password of the user attempting to log in.
+   * @return An {@link AuthResponse} containing the generated JWT token and user details.
+   */
+  @Override
+  public AuthResponse login(String email, String password) {
+    var authenticatedUser = authenticationService.checkUserCredentials(email, password);
+    String jwtToken = saveOrUpdateUserToken(authenticatedUser);
+    return createNewAuthResponse(jwtToken, authenticatedUser);
+  }
+
+  private AuthResponse createNewAuthResponse(String jwtToken, User user) {
+    var userDto = userMapper.mapToBasicDto(user);
+    return new AuthResponse(jwtToken, userDto);
+  }
+
+  /**
+   * Registers a new user in the system by validating the user data, saving the user,
+   * and generating a JWT token.
+   *
+   * <p>This method is transactional to ensure that user registration and token generation
+   * are performed atomically.
+   *
+   * @param user The {@link User} entity containing registration data.
+   * @return An {@link AuthResponse} containing the generated JWT token and user details.
+   * @throws IOException If an error occurs during user registration.
+   */
+  @Override
+  public AuthResponse register(UserDtoWithAvatarAndPassword request) {
+    log.info("register - {}", request);
+    var user = userMapper.mapToUserWithPassword(request);
+
+    var newUser = registerUser(createNewUser(user));
+    String jwtToken = saveOrUpdateUserToken(newUser);
+    return createNewAuthResponse(jwtToken, newUser);
+  }
+
+  private User createNewUser(User user) {
+    return User.builder()
+        .userName(user.getUserName())
+        .userEmail(user.getUserEmail().toLowerCase())
+        .password(user.getPassword())
+        .role(Role.USER)
+        .build();
+  }
+
+  @Override
+  public User getAuthenticatedUser() {
+    return authenticationService.getAuthenticatedUser();
+  }
+
+  /**
+   * Saves or updates a JWT token for the specified user. Any existing tokens associated with the user
+   * are deleted, and the new token is saved in the database.
+   *
+   * @param user The {@link User} entity for which the token is being generated.
+   * @return The generated JWT token as a string.
+   */
+  @Override
+  public String saveOrUpdateUserToken(User user) {
+    String jwtToken = tokenService.generateToken(user);
+    var token = createNewToken(jwtToken, user);
+    tokenService.deleteUserToken(user.getId());
+    tokenService.save(token);
+    return jwtToken;
+  }
+
+  @Override
+  public UserDetails getUserDetails(String token) {
+    Long userId = tokenService.extractId(token);
+    return userService.getUserDetails(userId);
+  }
+
+  /**
+   * Registers a new user by validating the user's registration data and saving the user entity.
+   *
+   * @param user The {@link User} entity containing registration data.
+   * @return The saved {@link User} entity.
+   */
+  @Override
+  public User registerUser(User user) {
```

Comment:
there should not be two public method with similar names `register` and `registerUser` . Maybe this could be private?